docker/Dockerfile

```
@@ -0,0 +1,60 @@
+# See the file "COPYING" in the main distribution directory for copyright.
+
+# Layer to build Zeek.
+FROM debian:buster-slim AS build_zeek
+
+# Configure system for build.
+RUN apt-get -q update \
+ && apt-get install -q -y --no-install-recommends \
+     bind9 \
+     bison \
+     cmake \
+     flex \
+     g++ \
+     gcc \
+     libmaxminddb-dev \
+     libpcap-dev \
+     libssl-dev \
+     libz-dev \
+     make \
+     python3-minimal \
+     python3-dev \
+     swig \
+     ninja-build \
+     python3-pip \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+# Copy over the Zeek source tree.
+# NOTE: This assumes that we build in the context of the parent directory (the
+# Git checkout root). We e.g., ensure that in the `Makefile` in this directory.
+COPY . /src/zeek
+RUN make -C /src/zeek distclean
+
+WORKDIR /src/zeek
+RUN ./configure \
+     --generator=Ninja \
+     --build-type=Release \
+ && ninja -C build install
```

```suggestion
 && ninja -C build -j $(nproc) install
```
I think the Github VMs are only two-core, but I did a quick test looking at their output of `ninja --help` and it suggests they don't parallelize by default. So this might help a bit.

---

Ninja should always parallelize by default (likely uses something like `$(nproc) + 1`), and that's also what I saw in the output when doing a quick check of `ninja --help` (showed `[default=3]`) in a `Dockerfile` with a `docker/build-push-action` action. Do you mean that Github suggested to not parallelize jobs?

---

Sigh ... I misread — I looked at the output of `-k`. You're right Benjamin, it already parallelizes (to 3).